source/session.ts

```diff
@@ -48,25 +48,29 @@ const ENCODE_DATETIME_FORMAT = "YYYY-MM-DDTHH:mm:ss";
  * @class  Session
  *
  */
-export class Session {
+export class Session<
+  TEntityTypeMap extends Record<string, any> = DefaultEntityTypeMap,
```

Comment:
I have a feeling that the extends `Record<string,any>` will break all the type safety, setting everything to any even if we specify a TEntityTypeMap

Reply:
you can look at the test/session.test.ts and toy with it a bit, seem to be working:
<img width="398" alt="image" src="https://github.com/ftrackhq/ftrack-javascript/assets/2225828/f4ab04a1-882b-4ad3-8603-6a18b14a5757">